.circleci/config.yml

```diff
@@ -64,6 +79,133 @@ jobs:
       - oss/npm_clean_install_with_caching
       - run:
           command: npm run codegen:check
+  Rust:
+    parameters:
+      platform:
+        type: executor
+    executor: << parameters.platform >>
+    environment:
+      # Because it's QUITE difficult to change the npm version on Windows,
+      # we'll use the version of npm that comes with this version of Node.js
+      # These versions can be found at https://nodejs.org/en/download/releases/.
+      NODE_VERSION: 14.16.1
+    steps:
+      - checkout
+      - when:
+          condition:
+            equal: [ *rust_macos_executor, << parameters.platform >> ]
+          steps:
+            - rust/install:
+                version: stable
+            - restore_cache:
+                keys:
+                  - rust-target-v1-macos-{{ checksum "Cargo.lock" }}
+            # Installing this with the curl nvm command was proving difficult,
+            # so I've opted to just install the .pkg.  Should be the same net
+            # result, even if liked the opportunity for symmetry with Linux.
+            - run:
+                name: Installing Node.js with a .pkg.
+                command: |
+                  curl "https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}.pkg" > "$HOME/Downloads/node.pkg" && sudo installer -store -pkg "$HOME/Downloads/node.pkg" -target "/"
+            - run:
+                name: Debug output of Node.js and npm versions
+                command: |
+                  echo 'Node.js version:'
+                  node --version
+                  echo 'npm version:'
+                  npm --version
+            - run: npm install
+            - rust/format:
+                cache_version: v1-macos
+            - rust/clippy:
+                with_cache: false # Only the first needs to restore cache.
```

Comment:
Hmm, this also means that only rust/format will *save* to the `~/.cargo` cache.  Does format actually do enough of the build for this to be worth it? (Maybe `rust/test` being first would be better, since I kinda assume test does more?) I guess those commands are designed to be run in parallel in separate jobs.

(Despite this comment, I think this PR as stands is a big improvement and have no interest in blocking merge.)

Reply:
I had thought `cargo fmt` did more of the compilation but it was `cargo clippy`.  I think it's worth moving them up, though my intent was originally to fail-fast for what was probably the most common thing to be missed locally: `fmt`-ing.

Reply:
There's also only one test.